internal/apiclient/config.go

```diff
@@ -0,0 +1,56 @@
+// internal/apiclient/config.go
+package apiclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// ClientConfig holds the client configuration values
+type ClientConfig struct {
+	Username   string `json:"username"`
+	BackendURL string `json:"backend_url"`
+}
+
+// LoadConfig reads configuration from a JSON file
+func LoadConfig(path string) (*ClientConfig, error) {
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+
+	file, err := os.Open(absPath)
+	if err != nil {
+		return nil, fmt.Errorf("error opening config file: %w", err)
+	}
+	defer file.Close()
+
+	bytes, err := io.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("error reading config file: %w", err)
+	}
+
+	var cfg ClientConfig
+	if err := json.Unmarshal(bytes, &cfg); err != nil {
+		return nil, fmt.Errorf("error parsing config file: %w", err)
+	}
+
+	if strings.TrimSpace(cfg.Username) == "" || strings.TrimSpace(cfg.BackendURL) == "" {
```

Comment:
should we add some validations?

Reply:
I added isValidURL that does basic checking against the URL, however the checking against username shouldn't be enforced IMO. specially, that we don't have a usersystem like git, we just allow the time entries to have a username info attached to it.